### PR TITLE
update s3 sensor docs to use context.cursor instead of context.last_run_key

### PR DESCRIPTION
**Pull request by prha**
### Summary & Motivation
We have better support for cursor than last_run_key, which should be deprecated.

### How I Tested These Changes
BK

### CODE REVIEW

**Comment by prha** [comment]
internal failure is unrelated.

**Review by gibsondan — DISMISSED** [review]
just one clarity concern, see inline

File: examples/docs_snippets/docs_snippets/concepts/partitions_schedules_sensors/sensors/sensors.py [hunk 1]
```diff
@@ -231,12 +231,14 @@ def multi_asset_sensor(context):
...
 def my_s3_sensor(context):
-    new_s3_keys = get_s3_keys("my_s3_bucket", since_key=context.last_run_key)
+    since_key = context.cursor or None
+    new_s3_keys = get_s3_keys("my_s3_bucket", since_key=since_key)
     if not new_s3_keys:
         yield SkipReason("No new s3 files found for bucket my_s3_bucket.")
         return
     for s3_key in new_s3_keys:
         yield RunRequest(run_key=s3_key, run_config={})
+        context.update_cursor(s3_key)
```
[the remark this answers is not in the document]
Comment (prha) [reply examples/docs_snippets/docs_snippets/concepts/partitions_schedules_sensors/sensors/sensors.py:241]: Okay, I think I'll just change the return signature (in a separate PR) to return a list of RunRequests.

Then the snippet will look like:
```
if not new_s3_keys:
    return SkipReason(...)

last_key = new_s3_keys[-1]
context.update_cursor(last_key)
return [RunRequest(...) for key in new_s3_keys]
```

File: examples/docs_snippets/docs_snippets/concepts/partitions_schedules_sensors/sensors/sensors.py [hunk 2]
```diff
@@ -231,12 +231,14 @@ def multi_asset_sensor(context):
 
 @sensor(job=my_job)
 def my_s3_sensor(context):
-    new_s3_keys = get_s3_keys("my_s3_bucket", since_key=context.last_run_key)
```
Comment (gibsondan) [inline examples/docs_snippets/docs_snippets/concepts/partitions_schedules_sensors/sensors/sensors.py:234]: nice - did you sanity check that this code runs (could even put it under test)

**Review by gibsondan — APPROVED** [review]

**Pull request merged**